mlrun/projects/project.py

```diff
@@ -868,6 +891,16 @@ def set_source(self, source, pull_at_runtime=False, workdir=None):
         """
         self.spec.load_source_on_run = pull_at_runtime
         self.spec.source = source or self.spec.source
+
+        if self.spec.source.startswith("git://"):
+
+            source, reference, branch = resolve_git_reference_from_source(source)
+            if not branch and not reference:
+                warnings.warn(
```

Comment:
is there a reason not to use our logger?

Reply:
does it reproduce different output when using `warnings.warn` vs `logger.warning`?

Reply:
right, `warnings.warn` is more for deprecation warnings while the `logger.warn` is more for a specific warning for the user. In our case is should be `logger.warn`.